File: src/baseDatos/Serializador.java
package baseDatos;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;
import java.util.ArrayList;

import gestorAplicacion.administracion.*;
import gestorAplicacion.usuario.*;

public class Serializador {

    private static void serializar(ArrayList<? extends Object> lista, String nombre) {
    
        File archivo = new File("");

        try{
            File path = new File(archivo.getAbsolutePath()+"/src/baseDatos/temp/"+nombre+".txt"); 
            
            FileOutputStream f = new FileOutputStream(path);
            ObjectOutputStream o = new ObjectOutputStream(f);
            
            o.writeObject(lista);

            o.close();
            f.close();
        }
        catch(FileNotFoundException e){
            System.out.println("No se encuentra el archivo");
        }
        catch(IOException e){
            System.out.println("Error Flujo de inicializacion");
        }
        // catch(ClassNotFoundException e){
        //     e.printStackTrace();
        // }
    }

    public static void serializarListas(){
        // Aqui van las clases que vamos a serealizar: ejem:
        Serializador.serializar(Estudiante.getEstudiantes(), "Estudiantes");
        Serializador.serializar(Grupo.getGruposTotales(), "Grupos");
        Serializador.serializar(Materia.materiasTotales, "Materias");
        Serializador.serializar(Usuario.getUsuariosTotales(), "Usuarios");
        Serializador.serializar(Coordinador.getCoordinadoresTotales(), "Coordinadores");
        Serializador.serializar(Horario.getHorariosTotales(), "Horarios");
        Serializador.serializar(Salon.salones, "Salones");
        Serializador.serializar(Beca.getBecas(), "Becas");
        Serializador.serializar(Profesor.getProfesores(), "Profesores");

        // y asi ...
    }


}

File: Python/src/baseDatos/Serializador.py
from gestorAplicacion.administracion.Beca import Beca
from gestorAplicacion.administracion.Grupo import Grupo
from gestorAplicacion.administracion.Horario import Horario
from gestorAplicacion.administracion.Materia import Materia
from gestorAplicacion.administracion.Salon import Salon
from gestorAplicacion.usuario.Coordinador import Coordinador
from gestorAplicacion.usuario.Estudiante import Estudiante
from gestorAplicacion.usuario.Profesor import Profesor
from gestorAplicacion.usuario.Usuario import Usuario
import pickle
import pathlib
import os


class Serializador:
    def serializar(nombre, lista):
        ruta = os.path.join(
            pathlib.Path(__file__).parent.absolute(), "temp/" + nombre + ".txt"
        )
        try:
            picklefile = open(ruta, "wb")
            pickle.dump(lista, picklefile)
            picklefile.close()
        except:
            print("Ocurrió un error")

    @classmethod
    def serializarDatos(cls):
        cls.serializar("Becas", Beca.getBecas())
        cls.serializar("Grupos", Grupo.getGruposTotales())
        cls.serializar("Horarios", Horario.getHorariosTotales())
        cls.serializar("Materias", Materia.getMateriasTotales())
        cls.serializar("Salones", Salon.getSalones())
        cls.serializar("Coordinadores", Coordinador.getCoordinadoresTotales())
        cls.serializar("Estudiantes", Estudiante.getEstudiantes())
        cls.serializar("Profesores", Profesor.getProfesores())
        cls.serializar("Usuarios", Usuario.getUsuariosTotales())
